osu.Server.Queues.ScoreStatisticsProcessor/Commands/Queue/WatchHighScoresCommand.cs

```diff
@@ -0,0 +1,228 @@
+using osu.Game.Rulesets;
+using osu.Server.QueueProcessor;
+using osu.Server.Queues.ScoreStatisticsProcessor.Helpers;
+
+namespace osu.Server.Queues.ScoreStatisticsProcessor.Commands.Queue
+{
+    /// <summary>
+    /// Watches for new high scores from the osu_scores_high tables and imports into the new scores table.
+    /// </summary>
+    /// <remarks>
+    /// This command is written under the assumption that only one importer instance is running concurrently.
+    /// This is important to guarantee that scores are inserted in the same sequential order that they originally occured,
+    /// which can be used for tie-breaker scenarios.
+    /// </remarks>
+    [Command("watch-high-scores", Description = "Imports high scores from the osu_scores_*_high tables into the new scores table.")]
+    public class WatchHighScoresCommand
+    {
+        /// <summary>
+        /// The ruleset to run this import job for.
+        /// </summary>
+        [Option(CommandOptionType.SingleValue, Template = "--ruleset-id")]
+        public int RulesetId { get; set; }
+
+        /// <summary>
+        /// When set to <c>true</c> and in watch mode, scores will not be queued to the score statistics processor,
+        /// instead being sent straight to the elasticsearch indexing queue.
+        /// </summary>
+        [Option(CommandOptionType.SingleOrNoValue, Template = "--skip-score-processor")]
+        public bool SkipScoreProcessor { get; set; }
+
+        [Option(CommandOptionType.SingleOrNoValue, Template = "--dry-run")]
+        public bool DryRun { get; set; }
+
+        private long lastCommitTimestamp;
+        private long startupTimestamp;
+
+        private ElasticQueuePusher? elasticQueueProcessor;
+        private ScoreStatisticsQueueProcessor? scoreStatisticsQueueProcessor;
+
+        /// <summary>
+        /// The number of seconds between console progress reports.
+        /// </summary>
+        private const double seconds_between_report = 2;
+
+        private ulong lastQueueId;
+
+        private readonly DateTimeOffset startedAt = DateTimeOffset.Now;
+
+        public async Task<int> OnExecuteAsync(CancellationToken cancellationToken)
+        {
+            Ruleset ruleset = LegacyRulesetHelper.GetRulesetFromLegacyId(RulesetId);
+            string highScoreTable = LegacyDatabaseHelper.GetRulesetSpecifics(RulesetId).HighScoreTable;
+
+            // When running in watch mode, we need to ascertain a few things:
+            // - There is active ongoing processing in `score_process_queue` – if not, there will be nothing to watch and we will switch to full run mode.
+            // - Check whether the full run (non-watch) has caught up to recent scores (ie. processed up to a `score_id` contained in `score_process_queue`,
+            //   which generally keeps ~3 days of scores). If not, we should warn that there is a gap in processing, and an extra full run should be performed
+            //   to catch up.
```

Comment:
How much of this is even relevant to this command at this point?

Notably the  "no ongoing processing" thing is reflected by an early-return below and I'm not sure that's expected behaviour from a pure watch command.

Reply:
Not relevant, will just nuke.